src/components/CreateCoursePage/index.jsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Helmet from 'react-helmet';
+import { Icon, StatusAlert } from '@edx/paragon';
+import { push } from 'connected-react-router';
+
+import CreateCourseForm from './CreateCourseForm';
+import store from '../../data/store';
+
+class CreateCoursePage extends React.Component {
+  componentDidMount() {
+    this.props.fetchOrganizations();
+  }
+
+  handleCourseCreate(options) {
+    const courseData = {
+      org: options.courseOrg,
+      title: options.courseTitle,
+      number: options.courseNumber,
+      mode: options.courseEnrollmentTrack,
+      price: options.coursePrice,
+    };
+    this.props.createCourse(courseData);
+  }
+
+  showErrorStatus(error) {
+    return (
+      <StatusAlert
+        id="error"
+        alertType="danger"
+        dismissible={false}
+        open
+        dialog={error}
+        className={['text-center', 'mx-auto', 'w-50']}
+      />
+    );
+  }
+
+  render() {
+    if (!this.props.publisherUserInfo) {
+      return this.showErrorStatus('Unable to load course creation form, please contact support');
+    }
+
+    if (this.props.courseInfo && this.props.courseInfo.courseCreated) {
+      store.dispatch(push(`/courses/${this.props.courseInfo.data.uuid}/`));
+    }
+
+    const {
+      courseOrg,
+      courseTitle,
+      courseNumber,
+      courseEnrollmentTrack,
+      coursePrice,
+      publisherUserInfo,
+      courseInfo,
+    } = this.props;
+
+    if (publisherUserInfo.error) {
+      return this.showErrorStatus(publisherUserInfo.error);
+    }
+
+    if (publisherUserInfo.isFetching) {
+      return (
+        <div className="mx-auto text-center">
+          <Icon
+            id="spinner"
+            className={['fa', 'fa-circle-o-notch', 'fa-spin', 'fa-3x', 'fa-fw']}
+          />
+        </div>
+      );
```

Comment:
We might want to pull this spinner construction into a component.

Reply:
I agree, should I do that in this PR you think?

Reply:
Done